src/sim/detail/InitializeTracks.cu

```diff
@@ -0,0 +1,339 @@
+                                   const TrackInitializerPointers inits)
+{
+    auto thread_id = KernelParamCalculator::thread_id().get();
+    if (thread_id < inits.vacancies.size())
+    {
+        // Get the track initializer, starting from the back of the vector
+        const TrackInitializer& init
+            = inits.initializers[inits.initializers.size() - thread_id - 1];
+
+        // Index of the empty slot to create the new track in
+        ThreadId slot_id(inits.vacancies[thread_id]);
+
+        // Initialize the simulation state
+        SimTrackView sim(states.sim, slot_id);
+        sim = init.sim;
+
+        // Initialize the particle physics data
+        ParticleTrackView particle(params.particle, states.particle, slot_id);
+        particle = init.particle;
+
+        // Copy the geometry state from the parent if possible
+        GeoTrackView geo(params.geo, states.geo, slot_id);
+        if (thread_id < inits.parent.size())
+        {
+            unsigned int parent_id
+                = inits.parent[inits.parent.size() - thread_id - 1];
+            GeoTrackView parent(params.geo, states.geo, ThreadId(parent_id));
+            geo.copy_state(parent, init.geo.dir);
+        }
+        // Initialize it from the position otherwise
+        else
+        {
+            geo = init.geo;
+        }
+    }
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Find empty slots in the track vector and count the number of secondaries
+ * that survived cutoffs for each interaction. If the track is dead and
+ * produced secondaries, fill the empty track slot with one of the secondaries.
+ */
+__global__ void locate_alive_kernel(const StatePointers            states,
+                                    const ParamPointers            params,
+                                    const TrackInitializerPointers inits)
+{
+    auto thread_id = KernelParamCalculator::thread_id().get();
+    if (thread_id < states.size())
+    {
+        // Secondary to copy to the parent's track slot if the parent has died
+        Secondary firstborn = Secondary::from_cutoff();
+
+        // Count how many secondaries survived cutoffs for each track
+        inits.secondary_counts[thread_id] = 0;
+        for (const auto& secondary : states.interactions[thread_id].secondaries)
+        {
+            if (secondary)
+            {
+                if (!firstborn)
+                {
+                    firstborn = secondary;
+                }
+                ++inits.secondary_counts[thread_id];
+            }
+        }
+
+        SimTrackView sim(states.sim, ThreadId(thread_id));
+
+        // The track is alive: mark this track slot as active
+        if (sim.alive())
+        {
+            inits.vacancies[thread_id] = flag_alive();
+        }
+        // The track is dead and produced secondaries: fill the empty track
+        // slot with the first secondary and mark the track slot as active
```

Comment:
Instead of adding visual separation between the `if` block and its sister `else`, move the comments in both cases inside the appropriate conditionals.